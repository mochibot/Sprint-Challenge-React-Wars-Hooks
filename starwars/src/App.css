.App {
  text-align: center;
}

/* write your parent styles in here for your App.js */

.AppHeader {
  background: linear-gradient( rgba(0, 0, 0, 0.4), rgba(0, 0, 0, 0.1));
} 