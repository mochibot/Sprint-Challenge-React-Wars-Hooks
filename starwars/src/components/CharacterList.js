import React from 'react';
import Character from './Character'

import { Container } from 'semantic-ui-react';

const CharacterList = (props) => {
  return (
    <Container fluid>
      {props.characters.map(item => <Character key={item.name} character={item}/>)}
    </Container>
  )
}

export default CharacterList;