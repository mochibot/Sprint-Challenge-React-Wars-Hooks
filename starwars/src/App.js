import React, { useState, useEffect } from 'react';
import axios from 'axios';
import { Header } from 'semantic-ui-react';

import Loading from './components/Loading'
import CharacterList from './components/CharacterList';


import 'semantic-ui-css/semantic.min.css'
import './App.css';

const App = () => {
  // Try to think through what state you'll need for this app before starting. Then build out
  // the state properties here.
  const [data, setData] = useState([]);
  // Fetch characters from the star wars api in an effect hook. Remember, anytime you have a 
  // side effect in a component, you want to think about which state and/or props it should
  // sync up with, if any.
  
  useEffect(() => {
    axios.get(`https://swapi.co/api/people/`)
      .then(response => {
        console.log('fetch data success: ', response)
        setData(response.data.results);
        console.log(data);
      })
      .catch(error => {
        console.log('error fetching data: ', error);
        console.log(!data);
        console.log(data);
      })
  }, [])

  return (
    <div className="App">
      <Header style={{fontSize: '30px', paddingTop: '10px'}}>React Wars</Header>
      {data.length === 0 ? <Loading/> : <CharacterList characters={data} />}
    </div>
  );
}

export default App;
