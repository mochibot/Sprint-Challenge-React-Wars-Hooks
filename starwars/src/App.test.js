import React from 'react';
import renderer from 'react-test-renderer';
import App from './App';
import CharacterList from './components/CharacterList';

describe('<App />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<App />).toJSON();

    expect(tree).toMatchSnapshot();
  });
})

describe('<CharacterList />', () => {
  it('should match snapshot', () => {
    const tree = renderer.create(<CharacterList />).toJSON();

    expect(tree).toMatchSnapshot();
  });
})