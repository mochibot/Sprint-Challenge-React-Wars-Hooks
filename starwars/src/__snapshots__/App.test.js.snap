// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<App /> should match snapshot 1`] = `
<div
  className="App"
>
  <div
    className="ui header"
    style={
      Object {
        "fontSize": "30px",
        "paddingTop": "10px",
      }
    }
  >
    React Wars
  </div>
  <div
    className="ui segment"
    style={
      Object {
        "padding": "40px",
      }
    }
  >
    <div
      className="ui active transition visible inverted dimmer"
      onClick={[Function]}
    >
      <div
        className="content"
      >
        <div
          className="ui inverted text loader"
        >
          Loading
        </div>
      </div>
    </div>
    <img
      className="ui image"
      src="/images/wireframe/short-paragraph.png"
    />
  </div>
</div>
`;
